tools/slang-test/main.cpp

```diff
@@ -74,9 +198,284 @@ struct Options
     Dictionary<TestCategory*, TestCategory*> excludeCategories;
 
     // By default we can test against all apis
-    int enabledApis = int(RenderApiFlag::AllOf);           
+    int enabledApis = int(RenderApiFlag::AllOf);
 };
-Options options;
+
+// Globals
+
+Options g_options;
+Dictionary<String, TestCategory*> g_testCategories;
+TestCategory* g_defaultTestCategory;
+
+// pre declare
+
+TestCategory* findTestCategory(String const& name);
+
+void append(const char* format, va_list args, StringBuilder& buf);
+
+void appendFormat(StringBuilder& buf, const char* format, ...);
+String makeStringWithFormat(const char* format, ...);
+
+/* !!!!!!!!!!!!!!!!!!!!!!!!!!!!! TestContext !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! */
+
+TestContext::TestContext(OutputMode outputMode):
+    m_outputMode(outputMode)
+{
+    m_totalTestCount = 0;
+    m_passedTestCount = 0;
+    m_failedTestCount = 0;
+    m_ignoredTestCount = 0;
+
+    m_inTest = false;
+    m_dumpOutputOnFailure = false;
```

Comment:
What is the relationship between `m_dumpOutputOnFailure` here and `dumpOutputOnFailure` in the `Options` type? I would have expected the `TestContext` to just hold a value of type `Options`, or to have the options-parsing code just directly modify a `TestContext`.